kcp/utils.sh

```diff
@@ -374,4 +333,34 @@ EOF
 
     KUBECONFIG="${CPS_KUBECONFIG}" kubectl apply -f "${SCRIPTPATH}"/../permissionToBindAPIExport.yaml
     rm permissionToBindAPIExport.yaml
+}
+
+createCredentialsForGitOpsEngineCluster() {
+  KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl create ns gitops || true
+  KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl create sa gitops-engine-cluster -n gitops || true
+  KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl create clusterrole gitops-engine-cluster --verb="*" --resource="*.*" || true
+  KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl create clusterrolebinding gitops-engine-cluster --clusterrole=gitops-engine-cluster --serviceaccount gitops:gitops-engine-cluster || true
+
+  KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: gitops-engine-cluster-token
+  namespace: gitops
+  annotations:
+    kubernetes.io/service-account.name: gitops-engine-cluster
+type: kubernetes.io/service-account-token
+EOF
+
+  host=$(KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl config view --minify -o jsonpath='{.clusters[0].cluster.server}')
+
+  while ! KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl describe secret gitops-engine-cluster-token | grep -E '^token' >/dev/null; do
```

Comment:
Missing `-n gitops` here. This line assumes it is the `gitops` namespace already.